scripts/eqtl_hail_batch/generate_eqtl_spearman.py

```diff
@@ -488,21 +488,15 @@ def spearman_correlation(df):
     t = hl.Table.from_pandas(spearman_df)
     t = t.annotate(global_bp=hl.locus(t.chrom, hl.int32(t.bp)).global_position())
     t = t.annotate(locus=hl.locus(t.chrom, hl.int32(t.bp)))
-    # get alleles
-    mt = mt.key_rows_by('locus')
-    t = t.key_by('locus')
-    t = t.annotate(
-        a1=mt.rows()[t.locus].alleles[0],
-        a2=hl.if_else(
-            hl.len(mt.rows()[t.locus].alleles) == 2, mt.rows()[t.locus].alleles[1], 'NA'
-        ),
-    )
     # add in vep annotation
-    t = t.annotate(functional_annotation=mt.rows()[t.locus].vep_functional_anno)
+    t = t.key_by('locus', 'alleles')
```

Comment:
This will cause a shuffle, but I guess there's no way around this, as you want to run the `spearman_correlation` function outside of Hail, which presumably can't be expressed easily as a Hail expression.

Reply:
Correct, this is one of the few instances where I use `key_by()`, but it's because the output of the association analysis is a df